src/main/java/com/conveyal/gtfs/stats/FeedStats.java

```diff
@@ -145,6 +154,40 @@ public LocalDate getEndDate() {
 
         return endDate;
     }
+
+    public LocalTime getStartTime (LocalDate date) {
+        return this.pattern.getStartTimeForTrips(getTripsForDate(date));
+    }
+
+    public LocalTime getEndTime (LocalDate date) {
+        return this.pattern.getEndTimeForTrips(getTripsForDate(date));
+    }
+
+    /** Get total revenue time (in seconds) for all trips on a given date. */
+    public int getTotalRevenueTimeForTrips (LocalDate date) {
+        return this.pattern.getTotalRevenueTimeForTrips(getTripsForDate(date));
+    }
+
+    /** Get total revenue distance (in meters) for all trips on a given date. */
+    public double getTotalRevenueDistanceForTrips (LocalDate date) {
+        return this.pattern.getTotalRevenueDistanceForTrips(getTripsForDate(date));
+    }
+
+    /** in seconds */
+    public double getDailyAverageHeadway (LocalDate date, LocalTime from, LocalTime to) {
+
+        OptionalDouble avg =  feed.stops.values().stream()
+                .map(s -> this.stop.getAverageHeadwayForStop(s.stop_id, date, from, to))
+                .mapToDouble(headway -> headway)
+                .average();
+
+        return avg.getAsDouble();
+    }
+
+    public double getAverageTripSpeed (LocalTime from, LocalTime to) {
```

Comment:
What are the units? Assuming furlongs per fortnight. Can we rename to getAverageTripSpeedKmh, or at least add some Javadoc to the method? (I generally prefer to put units right in the variables/function names that contain them so it's really clear, avoids Mars Climate Orbiter-type events).